### ISSUE TITLE
filter:post.email

### ISSUE THREAD

**Issue by julianlam** [NodeBB/NodeBB#7521]
RFC to determine whether it would be appropriate to add a new hook `filter:parse.email` for use in parsing... email html.

Right now posts when included in the digest (or post notifications, etc.) are parsed via `filter:parse.post`. As identified in #7277, this breaks in cases like the GitHub plugin, that pass through avatar info that is then not styled properly (as it relies on site stylesheet normally, which is not present in email).

**Comment by RoiEXLab** [comment]
One other thing to consider: I believe that formatting is also odd for previews inside custom widgets. So using such a hook for more than one thing might be worth a try.

**Comment by pitaj** [comment]
We should have specific filters for everything, and then a base filter for everything:

- parse.all
- parse.preview
- parse.teaser
- parse.post
- parse.email

**Comment by julianlam** [comment]
`parse.all` would be a little too easy for plugin devs to misuse, I think. Right now, you have to [explicitly combine the logic (as seen in mentions and markdown)](https://github.com/julianlam/nodebb-plugin-mentions/blob/master/library.js#L273) to multiple hooks if you want to use them.

If your filter transformation is truly capable of being applied consistently then it'd be safest to explicitly define it as such in the plugin. That is, it listens to all of the parse hooks and routes the requests to a single method handler.

**Comment by pitaj** [comment]
Fair enough, we should make the filter hold all have very similar arguments etc so one doesn't need to write separate functions for each, then.

**Comment by julianlam** [comment]
I have updated the title of the issue to `filter:post.plaintext` because I feel that more succinctly accomplishes the stated goal.

We can render a block of text like a post (with markdown, emoji, images, etc.), or plaintext, which would allow plugins to define separate handling for that use case.

So, for example, the raw text:

``` markdown
This is a post with [a markdown link](//example.org) and emoji :shipit: 

and a picture ![alt text](//example.org/picture.png)
```

Would render in plaintext as:

``` plaintext
This is a post with a markdown link and emoji [shipit emoji]

and a picture [alt text]
```

**Comment by julianlam** [comment]
One can make the argument that :smile: should be `:)` in plaintext, but that is outside the scope of this hook.

One could also make the argument that images would be better in emails, but it would be hard to determine when to insert the image and when not (plus sizing issues, etc.), so it seems safer to disallow images in UGC when rendered as plaintext.

**Comment by barisusakli** [comment]
Emojis and images work in latest emails, but yeah for things like rich embeds its 🎆 

![image](https://github.com/NodeBB/NodeBB/assets/2583394/9ceb35a6-d8b4-4901-a3c0-899181c6fd1c)

**Comment by mySebbe** [comment]
The requested capability is implemented through the existing hook rather than a new hook name. `Posts.parsePost(post, 'plaintext')` fires `filter:parse.post` with `type: 'plaintext'`, and parsers can branch on that value before Core applies plaintext sanitization. This preserves one parse pipeline while exposing the requested context, so this issue can be closed as implemented.
